proposals/lip-0067.md

```diff
@@ -83,36 +80,38 @@ In the following sections, we describe the uses of the properties of `keystoreSc
-
 ##### kdf and kdfparams
 
-The encryption key is an intermediate key derived from the user password. It is used to generate the secret key for decryption, and verify if the given password is correct. The function, and the params used to derive this key from the password are specified in `kdf`. The following values of `kdf` and `kdfparams` are allowed, depending on the key-derivation function:
+The encryption key is an intermediate key derived from the user password. It is used to generate the secret key for decryption, and verify if the given password is correct. Note that the encryption key is the leftmost 16 bytes of the derived key. The function, and the params used to derive this key from the password are specified in `kdf`. The following values of `kdf` and `kdfparams` are allowed, depending on the key-derivation function:
 
 | `kdf`            | function   | `kdfparams` | Definition |
 |------------------|------------|-------------|------------|
 | "PBKDF2-SHA-256" | `pbkdf2`   | `{iterations: uint32, salt: string}` | [RFC 2898](https://www.ietf.org/rfc/rfc2898.txt) |
 | "argon2id"       | `argon2id` | `{parallelism: uint32, iterations: uint32, memory: uint32, salt: string}` | [RFC 9106](https://datatracker.ietf.org/doc/html/rfc9106) |
 
-##### cipher and cipherparams
+##### mac
 
-The specified function encrypts the secret using the encryption key; to decrypt it, the encryption key along with `cipher` and `cipherparams` must be used. If the encryption key is longer than the key size required by the encoding function, it is truncated to the correct number of bits. The following option is supported:
+Computed as `SHA256(encryptionKey[-16:] + ciphertext)`, where `encryptionKey` is the second-leftmost 16 bytes of the key derived from the user password using the key-derivation function. The `mac` can be used to verify the encryption key before starting the decryption process.
 
-| `cipher`      | function      | `cipherparams` | Definition |
-|---------------|---------------|----------------|------------|
-| "AES-256-GCM" | `aes-256-gcm` | `{iv: string, tag: string}` | [RFC 5116](https://datatracker.ietf.org/doc/html/rfc5116#section-5.2) |
+#### id
+
+The `id` property stores a provided uuid (version 4 UUID as specified in [RFC 4122](https://datatracker.ietf.org/doc/html/rfc4122)), this is a randomly generated ID. It is used if the keystore needs to be referred to. Implementation help: ID generation is supported by nodejs with the [`uuid`](https://github.com/uuidjs/uuid#uuidv4options-buffer-offset) package, for example.
 
-Note that when using AES-256-GCM, the tag is an output of the encryption and is needed for decryption, this is why it is stored in the `cipherparams` property.
+#### version
+
+The `version` is set to `"3"`.
 
 #### metadata
 
```

Comment:
Comment about line 118: I know this text carries over from the past, but I think that what is saying can't be true. In particular, we say that the properties are not required and can be left empty; in LIP 64 however, we disallow having non-required properties. 

Again perhaps, I am missing something (e.g., might be that rules of LIP 64 do not apply for this LIP since it has to do with UI/UX related features and not protocol -- tagging @AndreasKendziorra @ricott1 who might know better) but seems to me that according to LIP 64, the JSON schema should be updated such that:

 - include "required" in "crypto", since it is of type "object".
 - I guess metadata should also be of type object? and have all properties required?

Reply:
As far as I know this schema is not used for Lisk Codec. If this is the case, then it's fine to have non-required properties. @ishantiw could you please confirm or disprove this?

Reply:
@ishantiw Can you please comment on Andreas's query to resolve this?

Reply:
Removed "metadata" from required properties.